check for elements existence in Cypress

handeyeco opened the pull request:
Two small sets of changes:

1. Several components that depended on TeX were breaking in Storybook because the components needed to be wrapped in `RenderStateRoot`, so I added that as a decorator.
2. Added a bunch of `.should("exist")` assertions in Cypress to help with retry-ability, but also to give us more happy green lines.

Full notes of my checkup at: https://khanacademy.atlassian.net/browse/LP-12482

EDIT: @nixterrimus I just realized I undid [some of the work you did](https://github.com/Khan/perseus/pull/276/files#diff-98b614e1838b171ee71c04450a8f1a562753193fb7d4f53a4de8b9b5a7e980ee) 🤦 Not intentional; I'll try to dig into this more so I understand why I needed it and you didn't need it.

jeremywiebe commented on packages/perseus/src/widgets/__tests__/grapher.cypress.jsx:
```diff
@@ -38,15 +38,19 @@ describe("Grapher widget", () => {
-                    .dragTo({x: left + 340, y: top + 140});
-            });
+            cy.get(GRAPHIE)
+                .should("exist")
```
Does this basically assert that the previous `.get()` returned an element?  If so, I love it!

jeremywiebe reviewed (approved):
I'm happy with this... thanks for making our tests more retryable and explicit.  

I did think as I was perusing if we'd want to have a `mustGet()` helper or something like that which combines `get(SELECTOR).should("exist")` but maybe that's not Cypressy.

handeyeco commented:
> @nixterrimus I just realized I undid [some of the work you did](https://github.com/Khan/perseus/pull/276/files#diff-98b614e1838b171ee71c04450a8f1a562753193fb7d4f53a4de8b9b5a7e980ee) 🤦 Not intentional; I'll try to dig into this more so I understand why I needed it and you didn't need it.

So what I think happened: `RenderStateRoot` was added to `RendererWithDebugUI` which wraps some stories but not all stories but seems only used for stories. I imagine this broke all stories that were wrapped with `RendererWithDebugUI` because stories were already wrapped in `RenderStateRoot` in `preview.js`; wrapping components twice causes the component to throw. To fix this, `RenderStateRoot` was removed from `preview.js`, but this broke components not wrapped in `RendererWithDebugUI`; components that need `RenderStateRoot` but aren't wrapped also throw.

I've removed `RenderStateRoot` from `RendererWithDebugUI` and added it to `preview.js`. All tests passed and I went through each component in Storybook to make sure they don't throw. I feel like I'm missing something though, @kevinbarabash could you check my logic on this?

kevinbarabash commented:
> I've removed `RenderStateRoot` from `RendererWithDebugUI` and added it to `preview.js`. All tests passed and I went through each component in Storybook to make sure they don't throw. I feel like I'm missing something though, @kevinbarabash could you check my logic on this?

I think that's fine.  I didn't realize there were some stories that weren't using `RendererWithDebugUI`.  preview.js seems like a good place for `RenderStateRoot`.   Thanks for making that change.

jeremywiebe reviewed (approved):
Great. Thanks for making these improvements Matthew!
